packages/smart-wallet/src/smartWallet.js

```diff
@@ -343,7 +343,9 @@ const behavior = {
      */
     async executeOffer(offerSpec) {
       const { facets, state } = this;
+      /** @type {State} */
       const {
+        address,
```

Comment:
hm. I missed something somewhere. A wallet knows the address it's associated with? Is that necessary? It sets off my ACL spidey-sense... it feels vagely like [caller](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/caller) or `msg.sender` in solidity.

Reply:
Yeah, I had the same tingle. It's not strictly necessary. We could take it out by passing two other things in instead:
- wallet logger
- wallet storageNode

But it has to persist through init and there may be some churn with durability, so I'll punt until then: https://github.com/Agoric/agoric-sdk/issues/5894